pkg/mdformatter/linktransformer/link.go

```diff
@@ -175,7 +178,30 @@ func NewValidator(logger log.Logger, linksValidateConfig []byte, anchorDir strin
 	v.c.OnError(func(response *colly.Response, err error) {
 		v.rMu.Lock()
 		defer v.rMu.Unlock()
-		v.remoteLinks[response.Ctx.Get(originalURLKey)] = errors.Wrapf(err, "%q not accessible; status code %v", response.Request.URL.String(), response.StatusCode)
+		retriesStr := response.Ctx.Get(retryKey)
+		retries, _ := strconv.Atoi(retriesStr)
+		switch response.StatusCode {
+		case 429:
+			if retries <= 2 {
+				response.Ctx.Put(retryKey, strconv.Itoa(retries+1))
+				retryAfter, convErr := strconv.Atoi(response.Headers.Get("Retry-After"))
+				if convErr == nil {
+					time.Sleep(time.Duration(retryAfter) * time.Second)
+				}
+
+				retryErr := response.Request.Retry()
+				v.remoteLinks[response.Ctx.Get(originalURLKey)] = errors.Wrapf(retryErr, "%q rate limited even after retry; status code %v", response.Request.URL.String(), response.StatusCode)
+			}
+		case 301, 307, 503:
+			if retries <= 2 {
```

Comment:
I set it to this in kind of yolo manner since I wasn't sure what a good default value would be. I was thinking this could be replaced by an option/flag later on!

Reply:
I set it to this in kind of yolo manner since I wasn't sure what a good default value would be. I was thinking this could be replaced by an option/flag later on!

Reply:
so what happens if we retry too many times? II can see we just return `v,nil` meaning all ok?

Reply:
A limit needs to be set so that it doesn't retry infinitely or too many times so I went with three but even if we did retry too many times there possibly wouldn't be any error just the statuscode would remain the same and take a long time. So I didn't return any error from there.

Reply:
Another maybe better approach could be retrying once and seeing if the result is the same as the previous statuscode, then error out. So we only retry once by default...